app/src/main/java/com/account/service/impl/AccountServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.account.service.impl;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Random;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.account.exception.ConflictException;
+import com.account.model.RequestModel;
+import com.account.model.ReturnModel;
+import com.account.model.ReturnModelResult;
+import com.account.persistence.entity.Account;
+import com.account.persistence.entity.AccountRole;
+import com.account.persistence.repository.AccountRepository;
+import com.account.persistence.repository.AccountRoleRepository;
+import com.account.service.AccountService;
+
+import jakarta.transaction.Transactional;
+
+@Service
+public class AccountServiceImpl implements AccountService {
+
+    @Autowired
+    private final AccountRepository accountRepository;
+
+    @Autowired
+    private final AccountRoleRepository accountRoleRepository;
+
+    public AccountServiceImpl(AccountRepository accountRepository, AccountRoleRepository accountRoleRepository) {
+        this.accountRepository = accountRepository;
+        this.accountRoleRepository = accountRoleRepository;
+    }
+
+    public String addRandomSuffixToUsername(String username) {
+        final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+        Random random = new Random();
+        StringBuilder sb = new StringBuilder(4);
+
+        for (int i = 0; i < 4; i++) {
+            int randomIndex = random.nextInt(CHARACTERS.length());
+            char randomChar = CHARACTERS.charAt(randomIndex);
+            sb.append(randomChar);
+        }
+
+        return username + "_" + sb;
+    }
+
+    public void validateData(RequestModel requestModel) {
+        String email = requestModel.getEmail();
+        String username = requestModel.getUsername();
+        Optional<Account> accountOptional = accountRepository.findByEmail(email);
+        if (accountOptional.isPresent())
+            throw new ConflictException("Email already exists!");
+        if (username == null) {
```

Comment:
move suffix generation to a separate method
https://en.wikipedia.org/wiki/Single-responsibility_principle